### DISCUSSION TITLE
How do I run Nextcloud AIO from source code?

### DISCUSSION THREAD

**Discussion by pepper-lees** [nextcloud/all-in-one#2351] [Questions] [answered]
Hi,

I'd like to play around with the front-end site code of the Nextcloud server and potentially contribute code, but I'm not quite sure how to set it up. Is it possible to take my existing Nextcloud install and switch it to running off source code (i.e. my own branch) instead of the beta/latest channels? If so, how do I do it? 

Keywords for search: developer mode, source code, build from source, contribution, code contribute

**Comment by szaimen** [comment, marked as answer]
Hi, I wouldn't recommend to use AIO for development purposes. For that better suited is https://github.com/juliushaertl/nextcloud-docker-dev. See https://cloud.nextcloud.com/s/iyNGp8ryWxc7Efa?path=%2F1%20Setting%20up%20a%20development%20environment

**Reply by pepper-lees** [reply]
Thanks!
